pkg/gofr/http/router.go

```diff
@@ -36,6 +47,109 @@ func (rou *Router) Add(method, pattern string, handler http.Handler) {
 	rou.Router.NewRoute().Methods(method).Path(pattern).Handler(h)
 }
 
+func (rou *Router) GetDefaultStaticFilesConfig() StaticFileConfig {
+	staticConfig := StaticFileConfig{
+		DirectoryListing: true,
+		HideDotFiles:     true,
+	}
+	return staticConfig
+}
+
+// Static File Handling.
+func (rou *Router) AddStaticFiles(endpoint string, staticConfig StaticFileConfig) {
+	fileServer := http.FileServer(http.Dir(staticConfig.FileDirectory))
+	rou.Router.NewRoute().PathPrefix(endpoint + "/").Handler(http.StripPrefix(endpoint, staticHandler(fileServer, staticConfig)))
+}
+
+// Check all the static handling configs.
+func staticHandler(fileServer http.Handler, staticConfig StaticFileConfig) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		url := r.URL.Path
+
+		const forbiddenBody string = "403 forbidden"
+
+		if staticConfig.DirectoryListing {
+			checkDirectoryListing(w, r, staticConfig, url)
+		}
+
+		filePath := strings.Split(url, "/")
+
+		fileName := filePath[len(filePath)-1]
+
+		if staticConfig.HideDotFiles {
+			checkDotFiles(w, r, staticConfig, fileName, url, forbiddenBody)
+		}
+
+		if len(staticConfig.ExcludeExtensions) > 1 {
+			checkExcludedExtensions(w, r, staticConfig, fileName, url, forbiddenBody)
+		}
+
+		if len(staticConfig.ExcludeFiles) > 1 {
+			checkExcludedFiles(w, r, staticConfig, fileName, url, forbiddenBody)
+		}
+
+		fileServer.ServeHTTP(w, r)
+	})
+}
+
+func checkDirectoryListing(w http.ResponseWriter, r *http.Request, staticConfig StaticFileConfig, url string) {
+	if _, err := os.Stat(filepath.Join(staticConfig.FileDirectory, "index.html")); err != nil && strings.HasSuffix(url, "/") {
+		http.NotFound(w, r)
+		return
+	}
+}
+
+func checkDotFiles(w http.ResponseWriter, r *http.Request, staticConfig StaticFileConfig, fileName string, url string, forbiddenBody string) {
+	if _, err := os.Stat(filepath.Join(staticConfig.FileDirectory, url)); err != nil {
+		http.NotFound(w, r)
+		return
+	}
+
+	if strings.HasPrefix(fileName, ".") {
+		w.WriteHeader(http.StatusForbidden)
+		w.Header().Set("Content-Type", "text/plain;charset=utf-8")
+		w.Write([]byte(forbiddenBody))
+
+		return
+	}
+}
+
+func checkExcludedExtensions(w http.ResponseWriter, r *http.Request, staticConfig StaticFileConfig, fileName string, url string, forbiddenBody string) {
+	if _, err := os.Stat(filepath.Join(staticConfig.FileDirectory, url)); err != nil {
+		http.NotFound(w, r)
```

Comment:
I would say it's not the role of this method

ServeHTTP would do it in a better (relative path resolution) and shared way

I would only return here, but also I would simply not do this if check

First for performance reason, because all your helpers are checking for the file existence.

Then, because it's not a problem to refuse all URLs containing a .pdf extension if you excluded them, even if the file doesn't exist.

A better way to do this, would be to use http.middleware in fact

Reply:
Couldn't implement with http.middleware but made the fixes mentioned above

Reply:
Yes, it's fine. It was just a suggestion